src/mode/graphics.rs

```diff
@@ -149,6 +215,21 @@ where
             return;
         }
 
+        // Keep track of max and min values
+        if (x as u8) < self.min_x {
+            self.min_x = x as u8;
+        }
+        if (x as u8) > self.max_x {
+            self.max_x = x as u8;
+        }
+
+        if (y as u8) < self.min_y {
+            self.min_y = y as u8;
+        }
+        if (y as u8) > self.max_y {
+            self.max_y = y as u8;
+        }
```

Comment:
These could be replaced with the shorter, cleaner:

```rust
self.min_x = self.min_x.min(x as u8);
self.max_x = self.max_x.max(x as u8);

self.min_y = self.min_y.min(y as u8);
self.max_y = self.max_y.max(y as u8);
```

The optimised assembly [appears to be largely the same](https://godbolt.org/z/yIY0bd) so I think performance is still good, but worth testing on your end to make sure. The unoptimised asm is pretty long so I don't know if the slowdown is acceptable in dev mode.